docs/cli.md

```diff
@@ -144,3 +159,39 @@ en.wikipedia.org
 en.wikipedia.org
 (---snip---)
 ```
+
+### Performance
+
+Our `adaparse` tool may outperform other popular alternatives. We offer a [collection of
+sets of URLs](https://github.com/ada-url/url-various-datasets) for benchmarking purposes.
+The following results are on a macBook Air 2022 (M2 processor) using LLVM 14. We
+compare against trurl version 0.6 (libcurl/7.87.0).
+
+
+On the wikipedia_100k dataset, we get that adaparse can generate normalized URLs about three
+times faster than trurl:
+
+```
+$ time cat url-various-datasets/wikipedia/wikipedia_100k.txt| trurl --url-file - &> /dev/null   1
+cat url-various-datasets/wikipedia/wikipedia_100k.txt  0,00s user 0,01s system 3% cpu 0,179 total
+trurl --url-file - &> /dev/null  0,14s user 0,03s system 98% cpu 0,180 total
+
+
+$ time cat url-various-datasets/wikipedia/wikipedia_100k.txt| ./build/tools/cli/adaparse -g href &> /dev/null
+cat url-various-datasets/wikipedia/wikipedia_100k.txt  0,00s user 0,00s system 10% cpu 0,056 total
+./build/tools/cli/adaparse -g href &> /dev/null  0,05s user 0,00s system 93% cpu 0,055 total
+````
+
+On the top100 dataset, the adaparse tool is twice as fast as the trurl:
+
+```
+$ time cat url-various-datasets/top100/top100.txt| trurl --url-file - &> /dev/null              1
```

Comment:
I recommend sharing these commands using `<summary><details>` markdown, and hide it. If user wants to see them they can click and expand on it.

More information: https://gist.github.com/scmx/eca72d44afee0113ceb0349dd54a84a2

Reply:
Yep.